### PR TITLE
[2.3][tests] Use @requires annotation when possible

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

### CODE REVIEW

File: src/Symfony/Bridge/Monolog/Tests/Processor/WebProcessorTest.php [hunk 1]
```diff
@@ -36,10 +36,6 @@ public function testUsesRequestServerData()
 
     public function testCanBeConstructedWithExtraFields()
     {
-        if (!$this->isExtraFieldsSupported()) {
-            $this->markTestSkipped('WebProcessor of the installed Monolog version does not support $extraFields parameter');
-        }
```
Comment (xabbuh) [inline src/Symfony/Bridge/Monolog/Tests/Processor/WebProcessorTest.php]: You can then also remove the `isExtraFieldsSupported()` method itself. However, doing this change requires to bump the dependency on `monolog/monolog` to `1.11.0` or higher (prior versions did not support extra fields in the web processor).

Comment (nicolas-grekas) [inline src/Symfony/Bridge/Monolog/Tests/Processor/WebProcessorTest.php]: then I'll revert

File: src/Symfony/Component/Form/Tests/AbstractLayoutTest.php [hunk 2]
```diff
@@ -22,7 +22,7 @@
     protected function setUp()
     {
         if (!extension_loaded('intl')) {
-            $this->markTestSkipped('The "intl" extension is not available');
+            $this->markTestSkipped('Extension intl is required.');
```
Comment (xabbuh) [inline src/Symfony/Component/Form/Tests/AbstractLayoutTest.php]: Can't we add the `@requires` annotation to the class level docblock instead?

Comment (nicolas-grekas) [inline src/Symfony/Component/Form/Tests/AbstractLayoutTest.php]: it's an abstract class and afaik annotations are not forwarded to sub-classes

File: src/Symfony/Component/Translation/Tests/Loader/XliffFileLoaderTest.php [hunk 3]
```diff
@@ -59,12 +59,11 @@ public function testIncompleteResource()
         $this->assertEquals(array('foo' => 'bar', 'extra' => 'extra', 'key' => '', 'test' => 'with'), $catalogue->all('domain1'));
     }
 
+    /**
+     * @requires extension mbstring
```
Comment (xabbuh) [inline src/Symfony/Component/Translation/Tests/Loader/XliffFileLoaderTest.php]: Don't you need to require the `iconv` extension too?

File: src/Symfony/Component/Validator/Tests/Constraints/LengthValidatorTest.php [hunk 4]
```diff
@@ -89,12 +89,11 @@ public function getFiveOrMoreCharacters()
         );
     }
 
+    /**
+     * @requires extension mbstring
```
Comment (xabbuh) [inline src/Symfony/Component/Validator/Tests/Constraints/LengthValidatorTest.php]: same here

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

File: src/Symfony/Component/Validator/Tests/Constraints/LengthValidatorTest.php (hunk 4 above)
Comment (nicolas-grekas) [inline src/Symfony/Component/Validator/Tests/Constraints/LengthValidatorTest.php]: only one of mbstring or iconv is required

Comment (xabbuh) [inline src/Symfony/Component/Validator/Tests/Constraints/LengthValidatorTest.php]: Well, but why skipping then if someone has the `iconv` but not the `mbstring` extension?

Comment (nicolas-grekas) [inline src/Symfony/Component/Validator/Tests/Constraints/LengthValidatorTest.php]: e don't care about tests portability, the only that counts are the continuously tested ones

**Pull request merged**